### PR TITLE
feat: deps update 08 2023

### PR DESCRIPTION
**Pull request by nikvoblikov**
**This PR:**
- Updates project dependencies.
- Updates tailwind config (As of Tailwind CSS v3.3 the line-clamp utilities are now included in the framework by default and this plugin is no longer required. [Link](https://github.com/tailwindlabs/tailwindcss-line-clamp))
- Updates mobile adaptation "NetworkPolicy Editor" blog post. [Original](https://cilium.io/blog/2021/02/10/network-policy-editor/) [After update](https://deploy-preview-338--cilium.netlify.app/blog/2021/02/10/network-policy-editor/);
- Adds YoutubeIframe component;

**Steps to test**

- Open the [All pages](https://deploy-preview-338--cilium.netlify.app/)
- Confirm that everything looks and works as expected

### CODE REVIEW

File: src/components/shared/popular-posts/popular-posts.jsx [hunk 1]
```diff
@@ -15,11 +15,10 @@ const PopularPosts = ({ className, titleTheme }) => {
     query BlogPagePopularPosts {
       allPopularPosts: allMdx(
         filter: {
-          fileAbsolutePath: { regex: "/posts/" }
+          internal: { contentFilePath: { regex: "/posts/" } }
           fields: { isPopular: { eq: true }, isFeatured: { eq: false } }
         }
         limit: 3
-        sort: { order: DESC, fields: fileAbsolutePath }
       ) {
```
Comment (vannyle) [inline src/components/shared/popular-posts/popular-posts.jsx:22]: Can it be sorted by contentFilePath?

Comment (vannyle) [reply src/components/shared/popular-posts/popular-posts.jsx:22]: If not, you can try to extract date and add it to fields using `onCreateNode`

File: src/posts/2020-11-10-cilium-19/index.md [hunk 2]
```diff
@@ -675,7 +658,6 @@ Feel free to ping us on [Slack] if you have any questions.
 [slack]: https://cilium.herokuapp.com/
 [getting started guides]: https://docs.cilium.io/en/v1.9/gettingstarted
 [upgrade guide]: https://cilium.readthedocs.io/en/v1.9/operations/upgrade/#upgrading-minor-versions
-[changelog]: https://github.com/cilium/cilium/blob/v1.9/CHANGELOG.md
```
Comment (vannyle) [inline src/posts/2020-11-10-cilium-19/index.md:678]: Is it intentional?

File: src/posts/26-06-2018-cilium-11/index.md [hunk 3]
```diff
@@ -385,7 +391,6 @@ Cilium deployment. Feel free to ping us on [Slack](http://cilium.io/slack).
 [mutual tls]: https://istio.io/docs/concepts/security/mutual-tls/
 [prometheus]: https://github.com/prometheus/prometheus
 [init container]: https://kubernetes.io/docs/concepts/workloads/pods/init-containers/
-[go 1.10]: https://blog.golang.org/go1.10
```
Comment (vannyle) [inline src/posts/26-06-2018-cilium-11/index.md:388]: Also here

File: src/templates/blog-post.jsx [hunk 4]
```diff
@@ -12,17 +12,24 @@ import MainLayout from '../layouts/main';
 const BlogPostPage = (props) => {
   const {
     data: { mdx: postData },
+    children,
     location: { pathname },
   } = props;
   const {
-    body: html,
```
Comment (vannyle) [inline src/templates/blog-post.jsx:18]: body should be removed in query below

File: src/utils/algolia-queries.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 const indexName = process.env.GATSBY_ALGOLIA_INDEX_NAME;
 
 const pageQuery = `{
-  pages: allMdx(filter: {fileAbsolutePath: {regex: "/posts/"}}) {
+  pages: allMdx(filter: { internal: { contentFilePath: {regex: "/posts/"}}}) {
```
Comment (vannyle) [inline src/utils/algolia-queries.js:4]: Could double-check if there is an excerpt in the query?

**Review by vannyle — COMMENTED** [review]
Good job, it is a massive work 💯

File: package.json [hunk 6]
```diff
@@ -20,71 +20,73 @@
     "prepare": "husky install"
   },
   "dependencies": {
-    "@headlessui/react": "^1.7.2",
-    "@hookform/resolvers": "^2.9.8",
+    "@headlessui/react": "^1.7.17",
+    "@hookform/resolvers": "^3.3.1",
     "@mdx-js/mdx": "^1.6.22",
```
[the remark this answers is not in the document]
Comment (vannyle) [reply package.json:24]: I meant this plugin seems @mdx-js/mdx

File: src/posts/2020-06-22-cilium-18/index.md [hunk 7]
```diff
@@ -958,4 +958,4 @@ to upgrade your Cilium deployment. Feel free to ping us on [Slack].
 
 [slack]: https://cilium.herokuapp.com/
 [publication]: https://dl.acm.org/doi/10.1145/3281411.3281443
-[ip fragmentation documentation]: https://docs.cilium.io/en/v1.8/concepts/networking/fragmentation
+[ip fragmentation documentation]: https://docs.cilium.io/en/v1.8/concepts/networking/fragmentatio
```
Comment (vannyle) [inline src/posts/2020-06-22-cilium-18/index.md:961]:
```suggestion
[ip fragmentation documentation]: https://docs.cilium.io/en/v1.8/concepts/networking/fragmentation
```

File: src/posts/2020-11-10-cilium-19/index.md [hunk 8]
```diff
@@ -1,5 +1,5 @@
 ---
-path: '/blog/2020/11/10/cilium-19'
+path: ' '
```
Comment (vannyle) [inline src/posts/2020-11-10-cilium-19/index.md:2]: ?

**Review by vannyle — APPROVED** [review]

File: src/posts/2020-11-10-cilium-19/index.md (hunk 2 above)
Comment (vannyle) [reply src/posts/2020-11-10-cilium-19/index.md:678]: Is that ok that the changelog is removed here?

File: src/posts/26-06-2018-cilium-11/index.md (hunk 3 above)
Comment (vannyle) [reply src/posts/26-06-2018-cilium-11/index.md:388]: It was marked as resolved, but why the link was removed?

File: src/posts/2020-11-10-cilium-19/index.md [hunk 9]
```diff
@@ -1,5 +1,5 @@
 ---
-path: ' '
+path: 'blog/2020/11/10/cilium-19'
```
Comment (vannyle) [inline src/posts/2020-11-10-cilium-19/index.md:2]:
```suggestion
path: '/blog/2020/11/10/cilium-19'
```

**Comment by xmulligan** [comment]
Let me know when this is ready to merge

**Comment by xmulligan** [comment]
@nikvoblikov is this ready to merge or do you still need more work?

**Comment by nikvoblikov** [comment]
> @nikvoblikov is this ready to merge or do you still need more work?

Hi! I'll make the final edit tomorrow, we'll check everything again and I'll ping you when we're ready.

**Comment by nikvoblikov** [comment]
@xmulligan Now it is ready to merge.

**Comment by xmulligan** [comment]
Great, thanks!

**Pull request merged**